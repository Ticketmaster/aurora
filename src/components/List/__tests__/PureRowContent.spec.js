import React from "react";
import renderer from "react-test-renderer";

import PureListRowContent from "../PureRowContent";

describe("<PureListRowContent />", () => {
  const modalRoot = global.document.createElement("div");
  modalRoot.setAttribute("id", "modal-root");
  const body = global.document.querySelector("body");
  body.appendChild(modalRoot);

  it("renders standard Pure Row Content correctly", () => {
    const component = renderer.create(
      <PureListRowContent index={0} onOverflowClick={jest.fn()}>
        content
      </PureListRowContent>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders standard Pure Row Content correctly with left triggers", () => {
    const component = renderer.create(
      <PureListRowContent
        index={0}
        rowTriggerPosition="left"
        onOverflowClick={jest.fn()}
      >
        content
      </PureListRowContent>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
