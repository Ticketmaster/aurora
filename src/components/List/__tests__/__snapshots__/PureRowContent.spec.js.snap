// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PureListRowContent /> renders standard Pure Row Content correctly 1`] = `
<div
  className="row__wrapper row__wrapper--collapsed dGIYBb"
>
  <div
    className="list__container fqHngb"
  >
    <div
      className="row__content dQsIEc"
    />
    <div
      className="cJSMnZ"
    >
      <button
        aria-label="More Info"
        className="button--more-info icon-button--last eBivpT"
        data-index={0}
        onClick={[MockFunction]}
        role="button"
        type="button"
      >
        <svg
          fill="rgba(38, 38, 38, 0.65)"
          height={22}
          viewBox="0 0 5 21"
          width={5.5}
        >
          <path
            d="M0 2.5C0 1.81.242 1.22.727.732A2.394 2.394 0 0 1 2.492 0a2.42 2.42 0 0 1 1.773.732C4.755 1.221 5 1.81 5 2.5s-.245 1.28-.735 1.768A2.42 2.42 0 0 1 2.492 5C1.8 5 1.212 4.756.727 4.268A2.418 2.418 0 0 1 0 2.5zm0 8.008c0-.692.242-1.283.727-1.773A2.39 2.39 0 0 1 2.492 8c.692 0 1.283.245 1.773.735S5 9.815 5 10.508a2.39 2.39 0 0 1-.735 1.765A2.43 2.43 0 0 1 2.492 13c-.692 0-1.28-.242-1.765-.727A2.403 2.403 0 0 1 0 10.508zM0 18.5c0-.69.242-1.28.727-1.768A2.394 2.394 0 0 1 2.492 16a2.42 2.42 0 0 1 1.773.732c.49.489.735 1.078.735 1.768s-.245 1.28-.735 1.768A2.42 2.42 0 0 1 2.492 21c-.692 0-1.28-.244-1.765-.732A2.418 2.418 0 0 1 0 18.5z"
          />
        </svg>
      </button>
    </div>
    <button
      aria-expanded={false}
      aria-label="Expand Row"
      className="button--expand-or-collapse button--collapsed bdgBvP"
      data-index={0}
      onClick={[Function]}
      role="button"
      size={45}
      type="button"
    >
      <svg
        fill="#1f262d"
        height={15}
        style={Object {}}
        viewBox="0 0 15 7"
        width={15}
      >
        <path
          d="M13.974.132a.614.614 0 0 0-.762 0L7.17 5.341 1.12.132C.916-.044.563-.037.376.125a.398.398 0 0 0-.167.332c0 .058.012.116.036.17a.385.385 0 0 0 .12.155l6.427 5.54a.568.568 0 0 0 .378.135.568.568 0 0 0 .377-.135l6.427-5.54a.422.422 0 0 0 .156-.325.42.42 0 0 0-.156-.325z"
        />
      </svg>
    </button>
  </div>
  <div
    className="container__overflow container__overflow--collapsed gNUlRn"
  >
    content
  </div>
</div>
`;

exports[`<PureListRowContent /> renders standard Pure Row Content correctly with left triggers 1`] = `
<div
  className="row__wrapper row__wrapper--collapsed hYrnSj"
>
  <div
    className="list__container fqHngb"
  >
    <button
      aria-expanded={false}
      aria-label="Expand Row"
      className="button--expand-or-collapse button--collapsed bdgBvP"
      data-index={0}
      onClick={[Function]}
      role="button"
      size={45}
      type="button"
    >
      <svg
        fill="#1f262d"
        height={15}
        style={Object {}}
        viewBox="0 0 15 7"
        width={15}
      >
        <path
          d="M13.974.132a.614.614 0 0 0-.762 0L7.17 5.341 1.12.132C.916-.044.563-.037.376.125a.398.398 0 0 0-.167.332c0 .058.012.116.036.17a.385.385 0 0 0 .12.155l6.427 5.54a.568.568 0 0 0 .378.135.568.568 0 0 0 .377-.135l6.427-5.54a.422.422 0 0 0 .156-.325.42.42 0 0 0-.156-.325z"
        />
      </svg>
    </button>
    <div
      className="row__content dQsIEc"
    />
    <div
      className="cJSMnZ"
    >
      <button
        aria-label="More Info"
        className="button--more-info icon-button--last eBivpT"
        data-index={0}
        onClick={[MockFunction]}
        role="button"
        type="button"
      >
        <svg
          fill="rgba(38, 38, 38, 0.65)"
          height={22}
          viewBox="0 0 5 21"
          width={5.5}
        >
          <path
            d="M0 2.5C0 1.81.242 1.22.727.732A2.394 2.394 0 0 1 2.492 0a2.42 2.42 0 0 1 1.773.732C4.755 1.221 5 1.81 5 2.5s-.245 1.28-.735 1.768A2.42 2.42 0 0 1 2.492 5C1.8 5 1.212 4.756.727 4.268A2.418 2.418 0 0 1 0 2.5zm0 8.008c0-.692.242-1.283.727-1.773A2.39 2.39 0 0 1 2.492 8c.692 0 1.283.245 1.773.735S5 9.815 5 10.508a2.39 2.39 0 0 1-.735 1.765A2.43 2.43 0 0 1 2.492 13c-.692 0-1.28-.242-1.765-.727A2.403 2.403 0 0 1 0 10.508zM0 18.5c0-.69.242-1.28.727-1.768A2.394 2.394 0 0 1 2.492 16a2.42 2.42 0 0 1 1.773.732c.49.489.735 1.078.735 1.768s-.245 1.28-.735 1.768A2.42 2.42 0 0 1 2.492 21c-.692 0-1.28-.244-1.765-.732A2.418 2.418 0 0 1 0 18.5z"
          />
        </svg>
      </button>
    </div>
  </div>
  <div
    className="container__overflow container__overflow--collapsed gNUlRn"
  >
    content
  </div>
</div>
`;
